File: 250429/요일 맞추기/guess-day-of-week.cpp
#include <iostream>
using namespace std;

int Days(int month, int day) {
    int days[13] = {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    int total_days = 0;
    for (int i = 1; i < month; i++) {
        total_days += days[i];
    }
    total_days += day;

    return total_days;
}
int main() {
    ios::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);

    int m1, d1, m2, d2;
    string weekdays[7] = {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
    cin >> m1 >> d1 >> m2 >> d2;

    int total_days = Days(m2, d2) - Days(m1, d1);
    while(total_days < 0) {
        total_days += 7;
    }
    cout << weekdays[total_days % 7];
    return 0;
}

File: 250429/요일 맞추기/guess-day-of-week.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int m1 = sc.nextInt(); 
        int d1 = sc.nextInt(); 
        int m2 = sc.nextInt(); 
        int d2 = sc.nextInt();
        String[] weekdays = new String[] { "Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun" };
        int[] daysInMonth = new int[] { 0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };

        boolean isReverse = false;
        if (m1 > m2 || (m1 == m2 && d1 > d2)) {
            int tempMonth = m1, tempDay = d1;
            m1 = m2;
            d1 = d2;
            m2 = tempMonth;
            d2 = tempDay;
            isReverse = true;
        }
        int totalDays = 0;
        if (m1 == m2) {
            totalDays = d2 - d1;
        } else {
            totalDays += daysInMonth[m1] - d1;
            for (int i = m1 + 1; i < m2; i++) {
                totalDays += daysInMonth[i];
            }
            totalDays += d2;
        }

        int startIndex = 0;
        int targetIndex = (startIndex + totalDays) % 7;

        if (isReverse) {
            targetIndex = (startIndex - totalDays % 7 + 7) % 7;
        }

        System.out.println(weekdays[targetIndex]);

        sc.close();
    }
}